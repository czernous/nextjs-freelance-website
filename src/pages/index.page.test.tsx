import { render, screen } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import '@testing-library/jest-dom';
import Home from './index.page';
import { act } from 'react-dom/test-utils';
import React from 'react';

expect.extend(toHaveNoViolations);

jest.mock('next/config', () => () => ({
  publicRuntimeConfig: {
    STRAPI_HOST: process.env.STRAPI_HOST,
  },
}));

describe('Home', () => {
  it('renders a heading (h1 - title)', () => {
    render(<Home />);

    const heading = screen.getAllByRole('heading');

    expect(heading[0]).toBeInTheDocument();
  });

  it('renders homepage unchanged', () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  test('has no axe violations', async () => {
    const { container } = render(<Home />);
    await act(async () => {
      expect(await axe(container)).toHaveNoViolations();
    });
  });
});
