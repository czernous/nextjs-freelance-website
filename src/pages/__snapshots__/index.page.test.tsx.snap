// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <div
    class="gradient"
  >
    <main
      class="main custom-container"
    >
      <h1
        class="title col-sm-10 col-md-11 col-lg-10"
      >
        Looking for awesome marketing?
      </h1>
      <h2
        class="subtitle col-sm-10 col-xl-9 col-xxl-11"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Aperiam ab esse culpa expedita nam aut officia quod consectetur corporis mollitia quidem totam aliquid error asperiores nihil, maiores dolorum debitis enim.
      </h2>
      <a
        class="button brick large shadow main-cta "
        href="https://www.google.com"
        target="_blank"
      >
        Get Quote
      </a>
    </main>
  </div>
</div>
`;
